docs/the-new-architecture/pillars-codegen.md

```diff
@@ -230,3 +230,59 @@ The **Codegen** for a Fabric Native Component contains a `MyFabricComponentManag
 Then, there is a layer of JNI C++ files that are used by Fabric to render the components. The basic element for a Fabric Component is the `ShadowNode`: it represents a node in the React abstract tree. The `ShadowNode` represents a React entity; therefore it could need some props, which are defined in the `Props` files and, sometimes, an `EventEmitter`, defined in the corresponding file.
 
 The **Codegen** also creates a `ComponentDescriptor.h`, which is required to get a proper handle on the Fabric Native Component.
+
+## The Codegen CLI
+
+**npx react-native codegen** [**--path** *path*] [**--platform** *string*] [**--outputPath** *path*]
+
+#### DESCRIPTION
+
+This command runs `react-native-codegen` for your project.
+The codegen must be configured as described in [Configure Codegen](../new-architecture-library-intro#configure-codegen).
+
+The following options are available:
+
+- `--path` - Path to `package.json`. The default path is current working directory.
+- `--platform` - Target platform. Supported values: `android`, `ios`, `all`. The default value is `all`.
+- `--outputPath` - Path to override `codegenCofig.outputDir`. The default value is `undefined`.
+
+#### EXAMPLES
+
+- Read `package.json` from the current working directory, generate code based on its `codegenConfig`.
+
+```sh
+npx react-native codegen
+```
+
+- Read `package.json` from the current working directory, generate iOS code in the location defined in the `codegenConfig`.
+
+```sh
+npx react-native codegen --platform ios
+```
+
+- Read `package.json` from `third-party/some-library`, generate Android code in `third-party/some-library/android/generated`.
+
+```sh
+npx react-native codegen \
+    --path third-party/some-library \
+    --platform android \
+    --outputPath third-party/some-library/android/generated
+```
+
+## Including Generated Code into Libraries
+
+If you are working on a library, you can include codegen artifacts in it. Here is how you can do it:
+
+- Define `includesGeneratedCode: true` and `outputDir` in the `codegenConfig`, as described in [Configure Codegen](../new-architecture-library-intro#configure-codegen).
+- Run codegen locally with [the codegen CLI](pillars-codegen#the-codegen-cli).
+- Update your `package.json` to include generated code.
+- Update your [podspec](pillars-turbomodules#ios-create-the-podspec-file) to include generated code.
+- Update your [build.gradle file](../the-new-architecture/pillars-turbomodules#the-buildgradle-file) to include generated code.
+
```

Comment:
Imho, we should specify that it is better to `gitignore` the generated files in the library.
Library author will always want to run Codegen before publishing.

Reply:
I think in this setup it is up to the library authors to decide wither they want to `gitignore` the generated code. There are pros and cons for both options.

Reply:
yeah, but we should provide a suggestion imho. As much as generated code is usually not checked in, to avoid unwilling file modifications and noise, I'd still add the suggestion to `gitignore` the generated files.